pkg/apis/core/types.go

```diff
@@ -2787,6 +2787,22 @@ type Sysctl struct {
 	Value string
 }
 
+// PodFSGroupChangePolicy holds policies that will be used for applying fsGroup to a volume
+// when volume is mounted.
+type PodFSGroupChangePolicy string
+
+const (
+	// OnRootMismatch indicates that volume's ownership and permissions will be changed
+	// only when permission and ownership of root directory does not match with expected
+	// permissions on the volume. This can help shorten the time it takes to change
+	// ownership and permissions of a volume.
+	OnRootMismatch PodFSGroupChangePolicy = "OnRootMismatch"
```

Comment:
Name it to "OnlyChangeOnRootMismatch".?

Reply:
I think because the type already says "ChangePolicy" in it, we don't need to say Change again in the string.

Reply:
Just feel "OnlyChangeOnRootMismatch" and "AlwaysChangeVolumeOwnership" (I think ownership is more accurate than permission?) match each other.

Or make it simpler "OnRootMismatch" and "Always" or "Force"?

Reply:
previous discussion about naming this field - https://github.com/kubernetes/enhancements/pull/1478#discussion_r371464115 . Bigger field names make for uglier UX.. :(

Reply:
ok the KEP actually use the one I mentioned above "OnRootMismatch" and "Always". So you want to use this?

Reply:
It already is using `OnRootMismatch`. Are you saying we change `AlwaysChangeVolumePermission` to just `Always`? This would result in some ambiguous type inside a common `core/types.go`

Reply:
So you mean, it will cause some errors if using "Always"?

Reply:
Not errors but a single "Always" string is confusing right? Like "Always" what? I would like to associate some additional context with the string "Always".

Like "PullAlways" says something is being pulled always (in k8s context that could mean an image)

Reply:
another example

	RestartPolicyAlways    RestartPolicy = "Always"
	RestartPolicyOnFailure RestartPolicy = "OnFailure"
	RestartPolicyNever     RestartPolicy = "Never"

Similar to this example, "Always" means always change no matter what. "OnRootMismatch" means only change if root level does not match. Like @msau, since the field name "FsChangePolicy" already indicate the action, no need to mentioned "change" again?

Reply:
okay you mean value `AlwaysChangeVolumePermission` variable right? It could be simply kept `"Always"` rather than `"AlwaysChangeVolumePermission"` - but the variable itself will be still called `AlwaysChangeVolumePermission` ?

Reply:
yes, I mean the value, not the variable name.

Reply:
Agree with Jing

Reply:
fixed.